### PR TITLE
Kill the TimerIdConv thread only for ruby 2.3.1 and lower.

### PR DESCRIPTION
**Pull request by jrafanie**
https://bugs.ruby-lang.org/issues/12342 has been fixed in 2.3.2+ and
2.4.0+ so we should only do this HACK for 2.3.1 and lower.

Extracted from https://github.com/ManageIQ/manageiq/pull/13104

### CODE REVIEW

File: lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb [hunk 1]
```diff
@@ -45,20 +45,20 @@ def teardown
...
+
       # This hack was done to prevent ruby from leaking the
       # TimerIdConv thread.
       # https://bugs.ruby-lang.org/issues/12342 (has been fixed in ruby 2.4.0 preview 1)
       # also fixed in ruby_2_3 branch for the 2.3.2 release: https://github.com/ruby/ruby/commit/c20b07d5357d7cb73226b149431a658cde54a697
-      thread = global_id_conv
-               .try(:instance_variable_get, '@holder')
-               .try(:instance_variable_get, '@keeper')
-      if RUBY_VERSION > "2.3.1"
-        warn "Remove me: #{__FILE__}:#{__LINE__} and my test. Ruby 2.3.2+ should not be creating timer threads."
```
Comment (gmcculloug) [inline lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb]: @jrafanie Did you take care of the "and my test" part?

- [X] Remove me
- [ ] and my test

Comment (jrafanie) [inline lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb]: I wasn't planning on removing the test here but now that you point it out... the test is pretty useless.  I'll remove it now.  Good riddance.

Comment (jrafanie) [inline lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb]: Updated @gmcculloug

**Comment by jrafanie** [comment]
marking as fine/yes, now that we support 2.3.6+ on fine branch

**Comment by jrafanie** [comment]
@miq-bot add_label fine/yes

**Comment by jrafanie** [comment]
@miq-bot rm_label fine/no

**Pull request merged**